tools/manual_testing/plot.ipynb

```diff
@@ -0,0 +1,571 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "525d4ea1-7fa3-469b-807d-cfc1b4b4f4a0",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import scipp as sc\n",
+    "import numpy as np\n",
+    "N = 10\n",
+    "M = 40\n",
+    "L = 30\n",
+    "K = 20\n",
+    "xx = np.arange(3, N, dtype=np.float64)\n",
+    "yy = np.arange(4, M, dtype=np.float64)\n",
+    "zz = np.arange(5, L, dtype=np.float64)\n",
+    "qq = np.arange(6, K, dtype=np.float64)\n",
+    "x, y, z, q = np.meshgrid(xx, yy, zz, qq, indexing='ij')\n",
+    "b = N/20.0\n",
+    "c = M/2.0\n",
+    "d = L/2.0\n",
+    "r = np.sqrt(((x-c)/b)**2 + ((y-c)/b)**2 + ((z-d)/b)**2  + ((q-d)/b)**2)\n",
+    "a = np.sin(r)\n",
+    "d = sc.Dataset()\n",
+    "da = sc.DataArray(data=sc.Variable(dims=['xx', 'yy', 'zz', 'Q_x'], values=a, unit='K'))\n",
+    "da.coords['xx'] = sc.Variable(dims=['xx'], values=xx)\n",
+    "da.coords['yy'] = sc.Variable(dims=['yy'], values=yy, unit='m')\n",
+    "da.coords['zz'] = sc.Variable(dims=['zz'], values=zz)\n",
+    "da.coords['Q_x'] = sc.Variable(dims=['Q_x'], values=qq)\n",
+    "sc.plot(da, projection='3d', aspect='auto')"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "70869384-c95b-425d-a325-8e4b32f729b3",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import scipp as sc\n",
+    "import numpy as np\n",
+    "N = 1000\n",
+    "M = 100\n",
+    "theta = np.random.random(N) * np.pi\n",
+    "phi = np.random.random(N) * 2.0 * np.pi\n",
+    "r = 10.0 + (np.random.random(N) - 0.5)\n",
+    "x = r * np.sin(theta) * np.sin(phi)\n",
+    "y = r * np.sin(theta) * np.cos(phi)\n",
+    "z = r * np.cos(theta)\n",
+    "\n",
+    "a = np.arange(M*N).reshape([M, N]) * np.sin(y)\n",
+    "pos = sc.vectors(dims=['xyz'], unit='m', values=np.array([x, y, z]).T)\n",
+    "da = sc.DataArray(\n",
+    "    data=sc.array(dims=['time', 'xyz'], values=a, unit='K'),\n",
+    "    #masks={\n",
+    "    #    'mask':pos.fields.x < 0.0 * sc.Unit('m')},\n",
+    "    coords={\n",
+    "        'xyz':pos,\n",
+    "        'time':sc.array(dims=['time'], values=np.arange(M).astype(np.float))})\n",
+    "# labels is hack so model does not drop our pos coord\n",
+    "# need panel so slices get defined in controller.update_axes\n",
+    "#da['time',0].plot(projection=\"3d\", positions='xyz', labels={'xyz':'xyz'})\n",
+    "da.plot(projection=\"3d\", positions='xyz')"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "Python 3",
+   "language": "python",
+   "name": "python3"
+  },
+  "language_info": {
+   "codemirror_mode": {
+    "name": "ipython",
+    "version": 3
+   },
+   "file_extension": ".py",
+   "mimetype": "text/x-python",
+   "name": "python",
+   "nbconvert_exporter": "python",
+   "pygments_lexer": "ipython3",
+   "version": "3.7.10"
+  }
+ },
+ "nbformat": 4,
+ "nbformat_minor": 5
+}
```

Comment:
One thing i noticed, I think the performance is much better when using a thick slice along a slider :+1: